src/Test/Utilities/Portable/Assert/AssertEx.cs

```diff
@@ -265,6 +265,98 @@ private static bool SequenceEqual<T>(IEnumerable<T> expected, IEnumerable<T> act
+                {
+                    var distance = levenshtein(item, expectedItem);
+                    if (distance < closestDistance)
+                    {
+                        closestDistance = distance;
+                        closestItem = expectedItem;
+                    }
+                }
+
+                if (closestItem != null)
+                {
+                    _ = indexes.TryGetValue(closestItem, out index);
+                    return index;
+                }
+
+                return -1;
+            }
+
+            // Adapted from Toub's https://blogs.msdn.microsoft.com/toub/2006/05/05/generic-levenshtein-edit-distance-with-c/
+            int levenshtein(string first, string second)
+            {
+                // Get the length of both.  If either is 0, return
+                // the length of the other, since that number of insertions
+                // would be required.
+                int n = first.Length, m = second.Length;
+                if (n == 0) return m;
+                if (m == 0) return n;
+
+                // Rather than maintain an entire matrix (which would require O(n*m) space),
+                // just store the current row and the next row, each of which has a length m+1,
+                // so just O(m) space. Initialize the current row.
+                int curRow = 0, nextRow = 1;
+                int[][] rows = new int[][] { new int[m + 1], new int[m + 1] };
+                for (int j = 0; j <= m; ++j) rows[curRow][j] = j;
+
+                // For each virtual row (since we only have physical storage for two)
+                for (int i = 1; i <= n; ++i)
+                {
+                    // Fill in the values in the row
+                    rows[nextRow][0] = i;
+                    for (int j = 1; j <= m; ++j)
+                    {
+                        int dist1 = rows[curRow][j] + 1;
+                        int dist2 = rows[nextRow][j - 1] + 1;
+                        int dist3 = rows[curRow][j - 1] + (first[i - 1].Equals(second[j - 1]) ? 0 : 1);
+                        rows[nextRow][j] = Math.Min(dist1, Math.Min(dist2, dist3));
+                    }
+
+                    // Swap the current and next rows
+                    if (curRow == 0)
+                    {
+                        curRow = 1;
+                        nextRow = 0;
+                    }
+                    else
+                    {
+                        curRow = 0;
+                        nextRow = 1;
+                    }
+                }
+
+                // Return the computed edit distance
+                return rows[curRow][m];
+            }
+        }
+
         public static void SetEqual<T>(IEnumerable<T> expected, IEnumerable<T> actual, IEqualityComparer<T> comparer = null, string message = null, string itemSeparator = "\r\n", Func<T, string> itemInspector = null)
         {
             var expectedSet = new HashSet<T>(expected, comparer);
```

Comment:
Shouldn't we be calculating a default here ...

Reply:
That's not necessary. Passing `null` is ok. 
That's existing logic and `HashSet` and `Dictionary` can handle `null` comparers.